### PR TITLE
Recipe Form and list

### LINKED ISSUE #8: Recipe List
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.

### PR DESCRIPTION
**Pull request by Aseadd**
## **Changes made**

close #8 
- Increase size of the text area
- Remove user text field
- Add styles

### CODE REVIEW

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class FoodsController < ApplicationController
 
   # GET /foods or /foods.json
   def index
-    @foods = Food.all
+    @foods = Food.find_by_sql("SELECT * FROM foods WHERE user_id = #{current_user.id}")
```
Comment (carlosmuerto) [inline app/controllers/foods_controller.rb:6]: avoid using SQL queries, this is vulnerable to SQL injection attacks, in this case, you can use 

``` ruby
  @food = current_user.food
```

current_user is a User model and the food come from the has_many

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 class RecipesController < ApplicationController
   before_action :set_recipe, only: %i[show edit update destroy]
+  before_action :authenticate_user!
 
   # GET /recipes or /recipes.json
   def index
-    @recipes = Recipe.all
+    @recipes = Recipe.find_by_sql("SELECT * FROM recipes WHERE user_id = #{current_user.id}")
   end
```
Comment (carlosmuerto) [inline app/controllers/recipes_controller.rb:8]: Same as food controller above

File: app/views/recipes/_form.html.erb [hunk 3]
```diff
@@ -13,22 +14,22 @@
...
   <div>
     <%= form.label :preparation_time_seconds, style: "display: block" %>
-    <%= form.number_field :preparation_time_seconds %>
+    <%= form.number_field :preparation_time_seconds, required: true, class: "form-input" %>
   </div>
 
   <div>
     <%= form.label :cooking_time_seconds, style: "display: block" %>
-    <%= form.number_field :cooking_time_seconds %>
+    <%= form.number_field :cooking_time_seconds, required: true, class: "form-input" %>
```
Comment (carlosmuerto) [inline app/views/recipes/_form.html.erb:27]: - the time in the database is stored in seconds to ensure consistency and easy conversion to other units, please ensure that the user can introduce time in a more user-friendly way.

File: config/routes.rb [hunk 4]
```diff
@@ -9,6 +9,11 @@
 
   resources :foods, only: [:index, :show, :new, :create, :destroy]
 
+  get '/user/foods', to: 'users/foods#index', as: :user_foods
+  get 'user/id', to: 'users#show', as: :user
+  # post '/user/foods/id', to: 'users/foods#index', as: :user_foods_path
+
```
Comment (carlosmuerto) [inline config/routes.rb:15]: - those routes are not used, please delete

**Review by carlosmuerto — CHANGES_REQUESTED** [review]
# Change required
## Great job, is looking great

- in the food list the table's head is not showing properly
  imagen

## consideration
if you have problems solving some bugs please fix the easiest one to merge right away and create an issue for each one.

File: app/views/recipes/_form.html.erb (hunk 3 above)
Comment (Aseadd) [reply app/views/recipes/_form.html.erb]: I make it users to prompt in minute

File: config/routes.rb (hunk 4 above)
Comment (Aseadd) [reply config/routes.rb]: Removed

File: app/controllers/recipes_controller.rb (hunk 2 above)
Comment (Aseadd) [reply app/controllers/recipes_controller.rb:8]: Done

**Review by carlosmuerto — APPROVED** [review]
# Approve
- there is some litter detailing to be solved, but is not merger blocking

**Pull request merged**